protobuf/lib/src/protobuf/message_set.dart

```diff
@@ -96,20 +95,18 @@ abstract class $_MessageSet extends GeneratedMessage {
             _parseExtension(typeId, message, extensionRegistry);
             typeId = null;
             message = null;
-            continue outer;
           }
         } else if (tagNumber == _messageSetItemMessageTag) {
           message = input.readBytes();
           if (typeId != null) {
             _parseExtension(typeId, message, extensionRegistry);
             typeId = null;
             message = null;
-            continue outer;
           }
         } else {
           // Skip unknown tags.
```

Comment:
Please add a comment that this will consume end-group marker.

Reply:
You can actually add an assertion after this loop that the last consumed tag was the 1+EGROUP

Reply:
I added the comment. We discussed the assertion offline, there are two ways that a message set message can end: with `EGROUP` (if it started with a `SGROUP`), or with end-of-input (when the message is at the top-level). So the assertion doesn't make sense (Java implementation [has this assertion][1] which I think is a bug).

I added a test to make sure we handle top-level message sets.

[1]: https://github.com/protocolbuffers/protobuf/blob/a95eb31ac45eed38a586a11cfb9c93faf765bafa/java/core/src/main/java/com/google/protobuf/GeneratedMessageLite.java#L856